### PR TITLE
Decrease commands to reduce layers

### LINKED ISSUE #1: Decrease the amount of Dockerfile commands
**What would you like to be added**:
After reviewing the Dockerfiles in this repo, I noticed that it uses a lot of Dockerfile commands for similar tasks (e.g. installing applications etc) which lead to new container layers. 

By combining comannds, you can decrease the amount of layers it creates.

**Why is this needed**:
Some design patterns suggest to keep the container layers as low as possible. We could use this approach to also update the Dockerfiles and add some more comments to it, so it's clear for externals what this containers contains and why it needs specific packages for example.

### PR DESCRIPTION
**Pull request by alee-cd**
**What this PR does / why we need it**:

This PR is combining commands to decrease the amount of layers of the image. And added comment to make it easier to read.

**Which issue(s) this PR fixes**:
Fixes #1

### CODE REVIEW

**Comment by ghost** [comment]
@alee-ntap Thank you for your contribution.

**Comment by Vincinator** [comment]
note: before merging, we need to squash the `WIP` commits in main

**Comment by nanory** [comment]
Hi @alee-ntap,
this looks already good but could you do the same also for `crossbuild.containerfile` so that it's streamlined?

File: build.containerfile [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install \
+       --no-install-recommends -y build-essential ca-certificates \
+       debhelper devscripts git sudo yq $(awk '{ print $1 }' pkgs) \
+    && apt-mark hold $(awk '{ print $1 }' pkgs) \
+    && gcc --print-search-dir && echo 'int main() { return 0; }' > main.c \
+    && gcc -o main main.c \
+    && ./main \
+    && find /tmp -mindepth 1 -delete
+# Copy the build scripts to /usr/local/bin
```
Comment (nanory) [inline build.containerfile:40]: Add a new line here

**Review by nanory — CHANGES_REQUESTED** [review]

**Comment by alee-cd** [comment]
> Hi @alee-ntap, this looks already good but could you do the same also for `crossbuild.containerfile` so that it's streamlined?

Ok. Will do.

**Review by nanory — APPROVED** [review]
LGTM

**Pull request merged**